modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItZoneInMemoryTest.java

```diff
@@ -136,30 +139,61 @@ private static List<CatalogStorageProfileDescriptor> extractZoneProfiles(Node no
 
         assertNotNull(zoneDescriptor);
 
-        return zoneDescriptor.storageProfiles().profiles();
+        List<CatalogStorageProfileDescriptor> storageProfilesDescripptors = zoneDescriptor.storageProfiles().profiles();
+
+        logStorageProfilesStatusMessage(node, storageProfilesDescripptors);
+
+        return storageProfilesDescripptors;
     }
 
-    private static boolean isRaftLogStorageVolatile(Node node, int zoneId) {
-        RaftServer raftServer = node.raftManager.server();
+    private static void logStorageProfilesStatusMessage(Node node, List<CatalogStorageProfileDescriptor> storageProfilesDescriptors) {
+        StringBuilder logMsgBuilder = new StringBuilder("Storage engines that are presented: {");
+
+        int lastStorageProfileIndex = storageProfilesDescriptors.size() - 1;
 
-        ConcurrentMap<RaftNodeId, RaftGroupService> nodes = getFieldValue(raftServer, JraftServerImpl.class, "nodes");
+        for (int storageProfileIdx = 0; storageProfileIdx <= lastStorageProfileIndex; storageProfileIdx++) {
```

Comment:
Please replace manual message building with log.info("... {}", expression-using-streams-producing-a-list), it will be a lot shorter and still easy to understand. Current code looks too complex. It even looks like you tried to optimize it, but performance is not an issue in tests

Reply:
Simplified.